pkg/decision/evaluator/condition.go

```diff
@@ -38,14 +39,23 @@ type ItemEvaluator interface {
 }
 
 // CustomAttributeConditionEvaluator evaluates conditions with custom attributes
-type CustomAttributeConditionEvaluator struct{}
+type CustomAttributeConditionEvaluator struct {
+	logger logging.OptimizelyLogProducer
+}
+
+// NewCustomAttributeConditionEvaluator creates a custom attribute condition
+func NewCustomAttributeConditionEvaluator(logger logging.OptimizelyLogProducer) *CustomAttributeConditionEvaluator {
+	return &CustomAttributeConditionEvaluator{logger: logger}
+}
 
 // Evaluate returns true if the given user's attributes match the condition
 func (c CustomAttributeConditionEvaluator) Evaluate(condition entities.Condition, condTreeParams *entities.TreeParameters) (bool, error) {
 	// We should only be evaluating custom attributes
 
 	if condition.Type != customAttributeType {
-		return false, fmt.Errorf(`unable to evaluator condition of type "%s"`, condition.Type)
+
+		c.logger.Warning(fmt.Sprintf(string(logging.UnknownConditionType), condition.StringRepresentation))
```

Comment:
Why string is needed? string(logging.UnknownConditionType)

Reply:
Since `UnknownConditionType` is of custom type, we must convert it to string before formatting.